pkg/apis/performance/v1alpha1/performanceprofile_types.go

```diff
@@ -55,11 +56,57 @@ type RealTimeKernel struct {
 	Enabled *bool `json:"enabled,omitempty"`
 }
 
+// PerformanceProfileConditionType valid conditions of a PerformanceProfile.
+type PerformanceProfileConditionType string
+
+const (
+	// PerformanceProfileUpdated means that the controller finished to update all targeted nodes.
+	PerformanceProfileUpdated PerformanceProfileConditionType = "Updated"
+
+	// PerformanceProfileUpdating means that at least one of machines targeted by performance profile still updating.
+	PerformanceProfileUpdating PerformanceProfileConditionType = "Updating"
+
+	// PerformanceProfileDegraded means that at least one of machines failed to finish update.
+	PerformanceProfileDegraded PerformanceProfileConditionType = "Degraded"
+)
+
+// PerformanceProfileCondition contains condition information for an PerformanceProfile.
+type PerformanceProfileCondition struct {
+	// Type of the condition, currently ('Updated', 'Updating', 'Degraded').
+	Type PerformanceProfileConditionType `json:"type"`
+
+	// Status of the condition, one of ('True', 'False', 'Unknown').
+	Status corev1.ConditionStatus `json:"status"`
+
+	// LastTransitionTime is the timestamp corresponding to the last status
+	// change of this condition.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// Reason is a brief machine readable explanation for the condition's last
+	// transition.
+	Reason string `json:"reason"`
+
+	// Message is a human readable description of the details of the last
+	// transition, complementing reason.
+	Message string `json:"message"`
+}
+
 // PerformanceProfileStatus defines the observed state of PerformanceProfile.
 type PerformanceProfileStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+	// MachineCount represents the total number of machines targeted by the performance profile.
+	MachineCount int32 `json:"machineCount"`
+
+	// UpdatedMachineCount represents the total number of machines that updated with parameters from the performance profile
+	// and ready for action.
+	UpdatedMachineCount int32 `json:"updatedMachineCount"`
+
+	// UnavailableMachineCount represents the total number of unavailable (non-ready) machines targeted by performance profile.
+	// A node is marked unavailable if it is in updating state or NodeReady condition is false.
+	UnavailableMachineCount int32 `json:"unavailableMachineCount"`
+
+	// conditions represents the latest available observations of current state.
+	// +optional
+	Conditions []PerformanceProfileCondition `json:"conditions,omitempty"`
```

Comment:
use structs in  ```github.com/openshift/custom-resource-status/conditions/v1```

there are also a bunch of helper functions in there like ```SetStatusCondition``` and ```IsStatusConditionTrue```

Reply:
done